libs/api/libbpf_link.cpp

```diff
@@ -15,7 +15,7 @@ bpf_link__pin(struct bpf_link* link, const char* path)
         return libbpf_err(-EBUSY);
     }
 
-    link->pin_path = strdup(path);
+    link->pin_path = ebpf_strdup(path);
```

Comment:
`strdup()` internally calls `malloc()`. Before the usersim changes, the user mode implementation of `ebpf_free()` called `free()`, which was correct, as it matched with the corresponding `malloc()` in strdup(). 

Now it looks like there is only one implementation of `ebpf_free()`, both for user mode and kernel mode, and it calls `ExFreePool()`. So, for user mode code also, we are invoking user sim implementation of `ExFreePool()`.

Is this expected?

Reply:
Until #2677 is done, yes this is expected.